build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/shared-configuration.kt

```diff
@@ -104,14 +106,36 @@ fun Project.addDependenciesAndConfigurations(prefix: String) {
     }
 }
 
+private
+fun CharSequence.kebabCaseToPascalCase() =
+    replace("-[a-z]".toRegex()) { it.value.drop(1).uppercase(Locale.US) }.uppercaseFirstChar()
 
 @Suppress("UnusedPrivateProperty")
 internal
 fun Project.addSourceSet(testType: TestType): SourceSet {
     val prefix = testType.prefix
     val sourceSets = the<SourceSetContainer>()
     val main by sourceSets.getting
-    return sourceSets.create("${prefix}Test")
+    val sourceSet = sourceSets.create("${prefix}Test")
+
+    val groovySourceDir = sourceSet.extensions.findByType<GroovySourceDirectorySet>()
+    // The task generate test class in Groovy, so it cannot be used if the project doesn't use Groovy for integration tests.
+    // This is the case for kotlin-dsl integration tests.
+    if (testType == TestType.INTEGRATION && groovySourceDir != null) {
+        val projectNameInPascalCase = name.kebabCaseToPascalCase()
+        val autoTestedSamplesTest = tasks.register<GenerateAutoTestedSamplesTestTask>("generateAutoTestedSamplesTest") {
+            mainSources.from(main.java)
+            output = layout.buildDirectory.dir("generated/sources/autoTested/groovy")
```

Comment:
💭 The `output` location seems like it should be a task convention set in the constructor.